### PR TITLE
Buildkite plugin to export EJSON secrets as environment variables

### PR DESCRIPTION
**Pull request by stevehodgkiss**
Exports secrets stored in an ejson file (default `.buildkite/secrets.ejson`) with `ejson2env`. This approach makes it easier for developers to add/change build secrets, see what secrets already exist. Secret changes would go through the PRs in this scenario, but they can't be changed outside the codebase like they can be currently.

### CODE REVIEW

**Review by orien — APPROVED** [review]
Excellent! 👏

Does this need a Buildkite pipeline of its own?

**Comment by stevehodgkiss** [comment]
@orien yep it sure does! I'll do that tomorrow.

The main issue I have right now is that secrets are exported after the repository's `pre-command` hook runs. Ideally it would export secrets before then, but I'm not sure if it's possible with the hooks currently available. We need to access files in the repo, so the `environment` hook won't work.

**Comment by orien** [comment]
Looks like there's `post-checkout` hook we could use.

https://buildkite.com/docs/agent/v3/hooks#available-hooks

**Comment by stevehodgkiss** [comment]
@orien that works! Thanks :)

File: plugin.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+author: https://github.com/envato
+requirements: []
+configuration:
+  properties:
+    ejson_file:
+      type: string
+    ejson_private_key_env_key:
+      type: string
+  required:
+    - ejson_file
```
Comment (gbakernet) [inline plugin.yml]: Required even though it has a default value?

**Comment by ali-graham** [comment]
👍

File: plugin.yml (hunk 1 above)
Comment (stevehodgkiss) [reply plugin.yml]: Addressed in bde927d

**Pull request merged**